services/account_service.go

```diff
@@ -31,14 +31,15 @@ func (s *AccountAPIService) AccountBalance(
 		return nil, ErrUnavailableOffline
 	}
 
+	blockchain := s.node.GetBlockchain()
```

Comment:
I think if you want to make this bulletproof then you need to do:
```
blockchain.ChainLock.RLock()
defer blockchain.ChainLock.RUnlock()
```

Then, you just need to test it to make sure the ChainLock isn't acquired e.g. in  BlockTip(). You should just inspect any of the `blockchain.` calls in these functions to make sure they're not acquiring locks, but also as long as it doesn't error you should be good.